math/scalapack/Portfile

```diff
@@ -28,8 +28,23 @@ github.tarball_from archive
 
 mpi.setup           require require_fortran
 
+# fix wrong version, which results in wrong versioned name of libscalapack.${version}.dylib
+# see https://github.com/Reference-ScaLAPACK/scalapack/issues/108
+post-patch {
+    reinplace "s|2.2.1|${version}|g" ${worksrcpath}/CMakeLists.txt
+}
+
 compiler.blacklist-append   {clang < 500}
 
+# remove duplicate rpath ${prefix}/lib/libgcc
+compilers.add_gcc_rpath_support no
+# remove duplicate rpath ${prefix}/lib
+if {${os.platform} eq "darwin" && ${os.major} >= 10} { # 10.6, Snow Leopard
```

Comment:
Maybe move this to its logical place, after configure block?

Reply:
Hm. The `pre-configure` and `compilers.allow_arguments_mismatch` blocks are also off order. Reorder them as well?

Reply:
Well, it is up to you and anyone who can merge it, and I think there are no defined rules in regard of ordering, it is just convenient to have a coherent order of stages, since it makes it easier to make sense of portfile or find something in it.

Reply:
Actually, having a look at the `pre-configure` block, I do not see why its `configure.args-append` must be enclosed by `pre-configure` and separate from the `configure.args` block. 
However, the reordering cosmetics somewhat obfuscate the essential code changes. Still go for it?

Reply:
Please leave MPICH handling as it is. Yes, it is intended to be that way.

No cosmetic changes are mandatory at all, you may just leave the PR as-is.

Reply:
I would have combined all of it simply to this:
```
configure.args      -DCMAKE_INSTALL_NAME_DIR=${prefix}/lib \
                    -DBUILD_SHARED_LIBS=ON \
                    -DBUILD_TESTING=ON \
                    -DMPI_C_COMPILER=${mpi.cc} \
                    -DMPI_Fortran_COMPILER=${mpi.fc} \
                    -DMPIEXEC_EXECUTABLE=${prefix}/bin/${mpi.exec} \
                    -DLAPACK_LIBRARIES="-L${prefix}/lib ${linalglib}"

```
A test did not reveal any problems.

Reply:
MPICH components may be picked before configure stage. I do not have time to test this port in every available configuration to ensure in this case it works as expected, but trust me, I have seen related failures, and several MPICH-using ports have a special configure logic for a reason.
“Do not touch something if it works” :) 

However, of course, anyone can do whatever they wish here, I cannot prevent merging anything, I will just move a port into my fork and use that.

Reply:
Waiting for reneeotten's feedback. As you are more experienced on mpi than me, I tend to follow your advice.

Reply:
BTW, this was not added by me into the port, it was originally designed this way: https://github.com/macports/macports-ports/commit/f6304fddfe06796941051db92b8567413479c897